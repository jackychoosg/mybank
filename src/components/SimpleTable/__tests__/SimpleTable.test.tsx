import { render, cleanup } from '@testing-library/react'
import { SimpleTable } from '../index'
import R from '../../../assets'

describe('SimpleTable component', () => {
  afterEach(cleanup)
  const columnKeys = {
    timestamp: R.strings.dashboard.tableHeader.timestamp,
    description: R.strings.dashboard.tableHeader.description,
    action: R.strings.dashboard.tableHeader.action,
    currency: R.strings.dashboard.tableHeader.currency,
    amount: R.strings.dashboard.tableHeader.amount
  }
  it('should take a snapshot with empty data', () => {
    const { asFragment } = render(
      <SimpleTable columnKeys={columnKeys} rows={[]} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should take a snapshot with data', () => {
    const data = [
      {
        accountId: 1,
        currencyId: 1,
        timestamp: 1625829468,
        reference: '948324085',
        amount: 1000,
        description: 'Testing',
        action: 'debit',
        id: 1
      }
    ]
    const { asFragment } = render(
      <SimpleTable columnKeys={columnKeys} rows={data} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
