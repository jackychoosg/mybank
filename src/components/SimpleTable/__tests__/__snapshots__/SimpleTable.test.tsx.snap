// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTable component should take a snapshot with data 1`] = `
<DocumentFragment>
  <table
    class="sc-bdnxRM frAAFw"
  >
    <thead
      class="sc-gtsrHT koYlQI st-header"
    >
      <tr
        class="sc-eCApnc"
      >
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Transaction Date
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Description
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Action
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Currency
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-jSFjdj st-body"
    >
      <tr
        class="sc-eCApnc"
      >
        <td
          class="sc-hKFxyN koVabA"
        >
          1625829468
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          Testing
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          debit
        </td>
        <td
          class="sc-hKFxyN koVabA"
        />
        <td
          class="sc-hKFxyN koVabA"
        >
          1000
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`SimpleTable component should take a snapshot with empty data 1`] = `
<DocumentFragment>
  <table
    class="sc-bdnxRM frAAFw"
  >
    <thead
      class="sc-gtsrHT koYlQI st-header"
    >
      <tr
        class="sc-eCApnc"
      >
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Transaction Date
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Description
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Action
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Currency
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-jSFjdj st-body"
    >
      <tr
        class="sc-eCApnc"
      >
        <td
          class="sc-hKFxyN koVabA"
          colspan="5"
        >
          <div
            class="sc-gKAaRy dTdeio"
          >
            There is no data, please choose other account
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
